adaptors/ampersand/base_collection.js

```diff
@@ -14,8 +16,110 @@ var eventBubbler = require('./event_bubbler');
  */
 var BaseCollection = AmpersandCollection.extend({
   tungstenCollection: true,
+  initialize: function() {
+    /* develblock:start */
+    this.initDebug();
+    /* develblock:end */
+    this.postInitialize();
 
+  },
+  /* develblock:start */
+
+  /**
+   * Bootstraps all debug functionality
+   */
+  initDebug: function() {
+    tungsten.debug.registry.register(this);
+    _.bindAll(this, 'getDebugName', 'getChildren');
+  },
+
+  /**
+   * Debug name of this object, using declared debugName, falling back to cid
+   *
+   * @return {string} Debug name
+   */
+  getDebugName: function() {
+    if (!this.cid) {
+      this.cid = _.uniqueId('collection');
+    }
+    return this.constructor.debugName ? this.constructor.debugName + this.cid.replace('collection', '') : this.cid;
+  },
+
+  /**
+   * Gets children of this object
+   *
+   * @return {Array} Whether this object has children
+   */
+  getChildren: function() {
+    return this.models;
+  },
+
+  /**
+   * Get a list of all trackable functions for this view instance
+   * Ignores certain base and debugging functions
+   *
+   * @param  {Object}        trackedFunctions     Object to track state
+   * @param  {Function}      getTrackableFunction Callback to get wrapper function
+   *
+   * @return {Array<Object>}                      List of trackable functions
+   */
+  getFunctions: function(trackedFunctions, getTrackableFunction) {
+    var result = [];
+    // Debug functions shouldn't be debuggable
+    var blacklist = {
+      constructor: true,
+      initialize: true,
+      postInitialize: true,
+      model: true,
+      initDebug: true,
+      getFunctions: true,
+      getVdomTemplate: true,
+      isParent: true,
+      getChildren: true,
+      getDebugName: true
+    };
+    for (var key in this) {
+      if (typeof this[key] === 'function' && blacklist[key] !== true) {
+        result.push({
+          name: key,
+          fn: this[key],
+          inherited: (key in BaseCollection.prototype)
+        });
+        this[key] = getTrackableFunction(this, key, trackedFunctions);
+      }
+    }
+    return result;
```

Comment:
Does this need the same Object.getOwnPropertyNames treatment that the Backbone version did?